src/ThaiRomanizationSharp.Thai2Rom/Thai2RomService.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using ThaiRomanizationSharp.Abstractions;
+using ThaiRomanizationSharp.Thai2Rom.Models;
+using ThaiRomanizationSharp.Thai2Rom.NeuralNet;
+using TorchSharp;
+using static TorchSharp.torch;
+
+namespace ThaiRomanizationSharp.Thai2Rom
+{
+    using static ThaiRomanizationSharp.Thai2Rom.Extensions;
+
+    /// <inheritdoc cref="IThaiRomanizationService"/>
+    public sealed class Thai2RomService : IThaiRomanizationService
+    {
+        private readonly Dictionary<string, int> charToIndex;
+        private readonly Dictionary<string, int> targetCharToIndex;
+        private readonly Dictionary<string, string> indexToTargetChar;
+        private readonly Seq2Seq network;
+
+        public Thai2RomService()
+            : this(Loader.LoadParameters()) { }
+
+        internal Thai2RomService(ModelParameters loader)
+        {
+            var (inputDim, eEmbDim, eHidDim, eDropout) = loader.EncoderParams;
+            var (outputDim, dEmbDim, dHidDim, dDropout) = loader.DecoderParams;
+
+            const int maxLength = 100;
+
+            this.charToIndex = loader.CharToIndex;
+            this.targetCharToIndex = loader.TargetCharToIndex;
+            this.indexToTargetChar = loader.IndexToTargetChar;
+
+            var encoder = new Encoder(inputDim, eEmbDim, eHidDim, eDropout);
+
+            var decoder = new AttentionDecoder(outputDim, dEmbDim, dHidDim, dDropout);
+
+            this.network = new Seq2Seq(
+                encoder,
+                decoder,
+                this.targetCharToIndex["<start>"],
+                this.targetCharToIndex["<end>"],
+                maxLength
+            );
+
+            this.network.load("Models/Data/model_state_dict");
+            this.network.Eval();
+        }
+
+        /// <inheritdoc cref="IThaiRomanizationService.Romanize(string)"/>
+        public string Romanize(string text)
+        {
+            if (text is null) throw new ArgumentNullException(nameof(text));
+            if (text is "") return "";
```

Comment:
At first I have thought if `text is string.empty` would work.

Reply:
Yep, only literals in patterns 🚮